File: BiBi/mandarin.js
function solution(k, tangerine) {
    const tans = {}
   tangerine.forEach((tan)=>{
    if(!tans[tan]){
        tans[tan]=0
    }
    tans[tan] +=1
   })


  const numberArr = Object.values(tans).sort((a,b) => b-a)

  let answer=0;
  let count = 0
  for(let i = 0; i<numberArr.length; i++){
    count += numberArr[i]
    if(count >= k){
        answer = i + 1
        break;
    }
  }
    return answer;
}

console.log(solution(6, [1, 3, 2, 5, 4, 5, 2, 3]))

File: Buzz/mandarin.java
import java.util.*;

class Solution {
    public int solution(int k, int[] tangerine) {
        
        int answer = 0;
        Map<Integer, Integer> map = new HashMap<>();
        for (int i : tangerine) {
            map.put(i, map.getOrDefault(i, 0) + 1);
        }

        List<Integer> list = (List<Integer>) map.values();

        // 최소
        list.sort(Collections.reverseOrder());

        int sum = 0;
        for (Integer integer : list) {
            answer++;
            sum += integer;
            
            if (sum >= k) {
                break;
            }
        }
        return answer;
    }
}